cli/main.rs

```diff
@@ -524,11 +530,21 @@ fn resolve_flags_and_init(
       ]
     }
   };
+  let is_single_threaded = !std::env::var("DENO_SINGLE_THREADED").unwrap_or_default().is_empty();
```

Comment:
Is this env var necessary? There's already a way to provide custom v8 flags and maybe the v8_platform created could be informed by the `--single-threaded` flag being provided?

Reply:
I think that's probably the right first step. I'll update this PR with that approach.

For some reason I had it in my mind that the http server ran on a separate thread (possibly only for http2 connections?) but I don't see any threads starting up when I make an http2 request with curl.

There are a several threads spawned for cache_db at startup but its probably easier to add an option to disable caching rather than try and make it switch to run in thread when single threading mode is active.